src/derived.rs

```diff
@@ -487,9 +549,9 @@ where
         key: &Q::Key,
     ) -> ProbeState<StampedValue<Q::Value>, MapGuard>
     where
-        MapGuard: Deref<Target = FxHashMap<Q::Key, QueryState<DB, Q>>>,
+        MapGuard: Deref<Target = QueryMap<DB, Q>>,
     {
-        match map.get(key) {
+        match map.data.get(key) {
```

Comment:
Instead of reaching into the `.data` field, we can add map api methods to `QueryMap`. However, we use all of `get`, `get_mut`, `insert`, `remove`, `iter`, `retain`, so I figure that that's to much delegation boilerplate